src/rantanplan/structures.py

```diff
@@ -11,230 +10,157 @@
+SILVA_LONG_LINE = LIRA_LONG_LINE
+SILVA_SHORT_LINE = LIRA_SHORT_LINE
+HENDECASYLLABLE = 11
+ALEXANDRINE = 14
+OCTOSYLLABLE = ARTE_MENOR_MAX_LENGTH
+TETRASYLLABLE = COPLA_ARTE_MENOR_MIN_LENGTH
+MINIMUM_SAFE_LENGTH = 3
+MAXIMUM_SAFE_LENGTH = 20
+#    STRUCTURES_LENGTH is a dictionary with a list of line lengths and a `True`
+#    or `False` flag to indicate if the structure can be repeated or not
+STRUCTURES_LENGTH = {
+    "seguidilla_compuesta": ([7, 5, 7, 5, 5, 7, 5], False),
+    "seguidilla": ([7, 5, 7, 5], False),
+    "chamberga": ([7, 5, 7, 5, 3, 7, 3, 7, 3, 7], False),
+    "seguidilla_gitana": ([6, 6, 11, 6], False),
+    "estrofa_safica": ([11, 11, 11, 5], False),
+    "estrofa_sáfica_unamuno": ([11, 11, 7, 5], False),
+    "estrofa_francisco_de_la_torre": ([11, 11, 11, 7], False),
+    "endecha_real": ([7, 7, 7, 11], True),
+    "lira": ([7, 11, 7, 7, 11], False),
+    "estrofa_manriqueña": ([8, 8, 4, 8, 8, 4], True),
+    "ovillejo_tetra": ([8, 4, 8, 4, 8, 4, 8, 8, 8, 8], False),
+    "ovillejo_tri": ([8, 3, 8, 3, 8, 3, 8, 8, 8, 8], False),
+    "haiku": ([5, 7, 5], False),
+    "sonnet": (14*[11], False),
+    "soleá": (3*[8], False),
+    "romance": ([8, 8, 8, 8], True),
+    "copla_real": (10*[8], False),
+    "espinela": (10*[8], False),
+    "copla_castellana": (8*[8], False),
+    "octava_real": (8*[11], False),
+    "cuaderna_vía": (4*[14], False),
+    "sexta_rima": (6*[11], False),
+}
 
 
-def most_common_lengths(lengths_list, n=4, number_of_verses=None):
-    """Return the n most commons lengths in a list of verses lengths
-
-    :param lengths_list: List with the lengths to be checked
-    :param n: Number of most common lengths to be returned
-    :param number_of_verses: How many verses are in a given stanza
-    :return: The n most common lengths
-    """
-    most_common_lengths_list = []
-    if len(lengths_list) == number_of_verses or number_of_verses is None:
-        count = Counter(lengths_list).most_common(n)
-        for length, _ in count:
-            # `lengths` is a tuple with the length and it's count so we only get
-            # the first element of the tuple
-            most_common_lengths_list.append(length)
-    return most_common_lengths_list
-
+def has_minimum_length(min_length, ranges_list):
+    """Checks if every range within a list of ranges contains a minimum length
 
-def is_seguidilla_compuesta(lengths):
-    """
-    [7, 5, 7, 5, 5, 7, 5]
-    :param lengths:
+    :param min_length: Minimum length
+    :param ranges_list: List of verse length ranges
     :return:
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 7:
-        for i in (0, 2, 5):
-            if 9 > lengths[i] > 5:
-                correct_lines += 1
-        for i in (1, 3, 4, 6):
-            if 7 > lengths[i] > 3:
-                correct_lines += 1
-    return correct_lines == number_of_verses
-
+    return all((min_length <= max(length_range) for length_range in ranges_list))
```

Comment:
You really want to keep the double parenthesis?

Reply:
I thought I removed that :scream: